src/sentry/web/frontend/auth_login.py

```diff
@@ -668,6 +668,22 @@ def handle_basic_auth(self, request: Request, **kwargs) -> HttpResponseBase:
                 if not user.is_active:
                     return self.redirect(reverse("sentry-reactivate-account"))
                 if organization:
+                    # Check if the user is a member of the provided organization based on their email
+                    membership = organization_service.check_membership_by_email(
+                        email=user.email, organization_id=organization.id
+                    )
+
+                    # If the user is a member, the user_id is None, and they are in a "pending invite acceptance" state with a valid invitation link,
+                    # we redirect them to the invitation page to explicitly accept the invite
+                    if (
+                        membership
+                        and membership.user_id is None
+                        and membership.is_pending
+                        and membership.invitation_link
```

Comment:
Because this is a new attribute, it won't exist during deploys and will raise runtime errors. You could avoid that by using `getattr(membership, 'invitation_link', None)` or deploying the new RPC attribute first.

Reply:
I did not know that! I will use the first option. Thanks!!